internal/tensorzero-node/src/napi_bridge.rs

```diff
@@ -0,0 +1,81 @@
+/// Macro to reduce boilerplate when calling methods defined in tensorzero-core from NAPI
+///
+/// All methods like this exposed to Node will return a string that Node client can deserialize into a response object. They may take no parameters, or take a string serialized from the corresponding JSON request object.
+///
+/// This macro handles:
+/// - Deserializing JSON parameters into a Rust struct
+/// - Calling the method defined in tensorzero-core
+/// - Serializing the result back to JSON
+/// - Error mapping
+///
+/// # Examples
+///
+/// ```ignore
+///     #[napi]
+///     pub async fn get_model_usage_timeseries(&self, params: String) -> Result<String, napi::Error> {
+///         napi_call!(
+///             &self,
+///             get_model_usage_timeseries,
+///             params,
+///             // Rust struct that represents the JSON type exposed to Node.
+///             GetModelUsageTimeseriesParams {
+///                 time_window,
+///                 max_periods
+///             }
+///         )
+///     }
+/// ```
+#[macro_export]
+macro_rules! napi_call {
```

Comment:
ideally this is heavily commented line by line so we can maintain easily. For those of us who aren't @Aaron1011 the macro syntax is a bit hard to read 😮‍💨